compiler/src/dmd/dsymbolsem.d

```diff
@@ -6726,13 +6730,28 @@ private void aliasAssignSemantic(AliasAssign ds, Scope* sc)
      */
 
     const errors = global.errors;
+    Dsymbol s;
+
+    // Try AliasSeq optimization
+    if (auto ti = ds.type.isTypeInstance())
+    {
+        if (!ti.tempinst.findTempDecl(sc, null))
+            return errorRet();
+        if (auto tempinst = isAliasSeq(sc, ti))
+        {
+            s = aliasAssignInPlace(sc, tempinst, aliassym);
+            if (!s)
+                return errorRet();
+            goto Lsymdone;
```

Comment:
I don't suppose its possible to avoid the use of goto here?

Reply:
Well, what are the options? a huge `else` block, copy pasting the epilogue,...

Reply:
Yeah, I was hoping you could chain those `if`s into a series of `else if`s. I was just checking to see if you had some bright idea.